components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -1030,9 +1030,29 @@ private void updateFederatedAuthenticatorConfig(FederatedAuthenticatorConfig new
             int authnId = getAuthenticatorIdentifier(dbConnection, idpId,
                     newFederatedAuthenticatorConfig.getName());
 
+            List<Property> unUpdatedProperties = new ArrayList<>();
             List<Property> singleValuedProperties = new ArrayList<>();
             List<Property> multiValuedProperties = new ArrayList<>();
 
+            // Checking for old fed auth config properties that are not updated so we can delete them.
+            if (ArrayUtils.isNotEmpty(oldFederatedAuthenticatorConfig.getProperties())) {
+                boolean isNewPropsEmpty = ArrayUtils.isNotEmpty(newFederatedAuthenticatorConfig.getProperties());
+                for (Property propertyOld : oldFederatedAuthenticatorConfig.getProperties()) {
+                    boolean hasProp = false;
+                    if (!isNewPropsEmpty) {
```

Comment:
Better check this before the above for loop.

Reply:
fixed